gammapy/modeling/models/core.py

```diff
@@ -681,6 +681,30 @@ def copy(self, copy_data=False):
             models=models, covariance_data=self.covariance.data.copy()
         )
 
+    def slice_by_energy(self, energy_min, energy_max, sum_over_energy_groups=False):
```

Comment:
This function should most likely not be public as it is an implementation detail specific to `TemplateNPredModel`. I would rather have it either private or a stand-alone function.